bluesky/magics.py

```diff
@@ -110,53 +113,161 @@ def ct(self, line):
+            except Exception:
+                prec = precision
+            value = np.round(v, decimals=prec)
             try:
-                values.append(p.position)
+                low_limit, high_limit = p.limits
             except Exception as exc:
-                values.append(exc)
-
-        headers = ['Positioner', 'Value', 'Low Limit', 'High Limit', 'Offset']
-        LINE_FMT = '{: <30} {: <11} {: <11} {: <11} {: <11}'
-        lines = []
-        lines.append(LINE_FMT.format(*headers))
-        for p, v in zip(positioners, values):
-            if not isinstance(v, Exception):
-                try:
-                    prec = p.precision
-                except Exception:
-                    prec = self.FMT_PREC
-                value = np.round(v, decimals=prec)
-                try:
-                    low_limit, high_limit = p.limits
-                except Exception as exc:
-                    low_limit = high_limit = exc.__class__.__name__
-                else:
-                    low_limit = np.round(low_limit, decimals=prec)
-                    high_limit = np.round(high_limit, decimals=prec)
-                try:
-                    offset = p.user_offset.get()
-                except Exception as exc:
-                    offset = exc.__class__.__name__
-                else:
-                    offset = np.round(offset, decimals=prec)
+                low_limit = high_limit = exc.__class__.__name__
             else:
-                value = v.__class__.__name__  # e.g. 'DisconnectedError'
-                low_limit = high_limit = offset = ''
+                low_limit = np.round(low_limit, decimals=prec)
+                high_limit = np.round(high_limit, decimals=prec)
+            try:
+                offset = p.user_offset.get()
+            except Exception as exc:
+                offset = exc.__class__.__name__
+            else:
+                offset = np.round(offset, decimals=prec)
+        else:
+            value = v.__class__.__name__  # e.g. 'DisconnectedError'
+            low_limit = high_limit = offset = ''
+
+        lines.append(LINE_FMT.format(p.name, value, low_limit, high_limit,
+                                     offset))
+    print('\n'.join(lines))
+
+
+def _labeled_devices(user_ns=None, maxdepth=6):
+    ''' Returns list of all devices that are labeled.
+
+        Parameters
+        ----------
+        user_ns : dict, optional
+            The namespace to search on
+
+        maxdepth: int, optional
+            max recursion depth
+
+        Returns
+        -------
+            A dictionary of (name, ophydobject) tuple indexed by device label.
+
+        Examples
+        --------
+        Read devices labeled as motors:
+            objs = _labeled_devices()
+            my_motors = objs['motors']
+    '''
+    # could be set but lists are more common for users
+    obj_list = collections.defaultdict(list)
+
+    if maxdepth <= 0:
+        print("Recursion limit exceeded")
```

Comment:
Cool, seems reasonable to give up at a certain point rather than raise. I would prefer a warning here so that i can be tested and, if desired, suppressed.

Reply:
sounds good. i used warnings.warn